scripts/content.js

```diff
@@ -0,0 +1,115 @@
+// Final content HTML element. Unfinished, as is the constructContent method
+const content = {
+    innerHTML: "",
+}
+
+// Class that organizes content data. This is then turned into the final HTML element.
+class contentData {
+    constructor(title, body, images) {
+        this.title = title;
+        this.body = body;
+        this.images = images;
+        this.timestamp = firebase.firestore.Timestamp.now();
+    }
+}
+
+class contentDB {
+
+    collection = "content"
+
+    // Converts content data into a normal object which can be stored.
+    converter = {
+        toFirestore: function(data) { // Converts to firestore data format when writing
+            return {
+                title: data.title,
+                body: data.body,
+                images: data.images,
+                timestamp: data.timestamp
+                };
+        },
+        fromFirestore: function(snapshot, options){ // Converts from firestore data format when reading
+            const data = snapshot.data(options);
+            return new contentData(data.title, data.body, data.images, data.timestamp);
+        }
+    }
+
+    // Gets content HTML for a note.
+    async getContent(contentID) {
+        var contentData = await this.getContentData(contentID);
+
+        // Still need to write method for contentData --> HTML
+        // noteContent = this.constructContent(contentData);
+        return contentData;
+    }
+
+    // Gets content data.
+    getContentData(contentID) {
+        var docRef = db.collection(this.collection).doc(contentID)
+        var contentData = docRef.withConverter(this.converter).get().then(doc => {
+            if (doc.exists) {
+                return doc.data();
+            } else {
+                console.log("No such document!");
+            }
+        })
+
+        return contentData;
+    }
+
+    // UNFINISHED, transforms content data into an HTML element. 
+    constructContent(contentData) {
+        // construct content HTML from content data
+        // return content HTML
+    }
+
+    // Writes content data.
+    async setContent(contentData) {
```

Comment:
Does it include everything for Content or just the Id?

Reply:
Includes everything. Basically writes the title, body, images, and timestamp to firebase and returns the ID of where the data was written to.